gen/nvml/lib.go

```diff
@@ -66,8 +72,8 @@ func (l *library) Lookup(name string) error {
 }
 
 // newDynamicLibrary is a function variable that can be overridden for testing.
-var newDynamicLibrary = func(name string, flags int) dynamicLibrary {
-	return dl.New(name, flags)
+var newDynamicLibrary = func(path string, flags int) dynamicLibrary {
+	return dl.New(path, flags)
```

Comment:
I guess the `dl` library is already capable of handling a full path, we just never used this until now?

Reply:
Yes. When `dl.Open` is called, the `path` is forwarded to `C.dlopen` which has well defined semantics for handling the path and where to search.